src/platform/plugins/shared/content_management/server/routes/list/query_builder.ts

```diff
@@ -0,0 +1,328 @@
+
+/**
+ * Validates that a field name is safe for use in Elasticsearch queries and runtime mappings.
+ * Field names must start with a letter and contain only alphanumeric characters and underscores.
+ * This prevents script injection vulnerabilities when field names are interpolated into Painless scripts.
+ *
+ * @param fieldName - The field name to validate.
+ * @returns `true` if the field name is valid, `false` otherwise.
+ *
+ * @example
+ * isValidFieldName('title'); // true
+ * isValidFieldName('my_field'); // true
+ * isValidFieldName('_private'); // false (must start with letter)
+ * isValidFieldName('field.nested'); // false (dots not allowed)
+ */
+export const isValidFieldName = (fieldName: string): boolean => {
+  return VALID_FIELD_NAME_REGEX.test(fieldName);
+};
+
+/**
+ * Kbn config-schema validator for field names used in sort and attribute operations.
+ * Enforces the same rules as {@link isValidFieldName}: must start with a letter and
+ * contain only alphanumeric characters and underscores.
+ *
+ * Used in route request validation to ensure user-provided field names are safe
+ * before they are used in Elasticsearch queries or runtime mappings.
+ *
+ * @example
+ * // In route validation:
+ * schema.object({
+ *   sortField: fieldNameSchema,
+ * })
+ */
+export const fieldNameSchema = schema.string({
+  minLength: 1,
+  maxLength: 100,
+  validate: (value) => {
+    if (!isValidFieldName(value)) {
+      return 'Field name must start with a letter and contain only alphanumeric characters and underscores';
+    }
+  },
+});
+
+/**
+ * Builds Elasticsearch runtime mappings for sortable fields.
+ *
+ * Creates a `sortable_title` runtime field that extracts the title from the type-specific
+ * attributes in the document source. Optionally creates an additional runtime field for
+ * custom sort fields (e.g., `sortable_myField`).
+ *
+ * The runtime mappings use Painless scripts to access attributes stored at `source[type]`
+ * (not `source[type].attributes`), matching Kibana's saved object document structure.
+ *
+ * @param sortField - Optional custom field name to create a runtime mapping for.
+ *   Root fields (`updatedAt`, `createdAt`, `updatedBy`, `createdBy`, `managed`) and
+ *   `title` are handled specially and don't need additional runtime mappings.
+ * @returns Elasticsearch runtime field mappings.
+ *
+ * @example
+ * // For sorting by title:
+ * buildRuntimeMappings(); // Returns { sortable_title: {...} }
+ *
+ * @example
+ * // For sorting by a custom attribute:
+ * buildRuntimeMappings('version'); // Returns { sortable_title: {...}, sortable_version: {...} }
+ */
+export const buildRuntimeMappings = (sortField?: string): estypes.MappingRuntimeFields => {
+  // Known root fields that don't need runtime mappings for sorting.
+  const rootFields = ['updatedAt', 'createdAt', 'updatedBy', 'createdBy', 'managed'];
+
+  const mappings: estypes.MappingRuntimeFields = {
+    sortable_title: {
```

Comment:
Could it be that the better way to approach this is to make "title" sortable in all supported saved objects?

Also, does this runtime approach plays nicely if title is already sortable? I think you mentioned that "maps" has it sortable already

Reply:
Opposite-- maps doesn't have the keyword mapping.

I'll see what logic I can add.

Reply:
Looking into this, I found the following:

> 1. Most saved object types use `title: { type: 'text' }` without keyword (dashboard, maps, search, visualization, etc.)
> 2. The runtime field approach works uniformly for all types
> 3. The performance impact of runtime fields is minimal for listing operations

The logic to determine if the runtime field is necessary would vary by object type, and require interrogating a number of mappings.

Short of hard-coding _exceptions_ to needing the field, I'm inclined to agree that this can just be the de-facto approach.

Your thoughts?